src/com/dabomstew/pkrandom/romhandlers/Gen4RomHandler.java

```diff
@@ -4243,4 +4243,42 @@ public BufferedImage getMascotImage() {
             throw new RandomizerIOException(e);
         }
     }
+
+    @Override
+    protected void randomizeHeldItems(Trainer t, Settings settings) {

```

Comment:
The randomization logic for this shouldn't be in `GenXRomHandler` - it should be handled in `AbstractRomHandler`. The method in `AbstractRomHandler` should extract the data using the generation specific `getTrainers`, then perform the generation agnostic randomization logic, then set the new data using the generation specific `setTrainers`. 

To access things like `GenXConstants.consumableHeldItems` in `AbstractRomHandler`, create a method in `RomHandler` that extracts this in a generation specific way (i.e. `getConsumableHeldItems`, and then you implement it in `Gen4RomHandler` as `return Gen4Constants.consumableHeldItems`, etc.)

This whole method can just be incorporated into `randomizeTrainerHeldItems` in `AbstractRomHandler`, it's not really long enough to warrant a sub-method.

Reply:
Done.